packages/web-components/fast-html/src/components/utilities.ts

```diff
@@ -514,40 +553,85 @@ export interface ChainedExpression {
     next?: ChainedExpression;
 }
 
+/**
+ * Evaluates parts of an expression chain
+ * @param parts Each part of an expression chain
+ * @param operator The operator between expressions
+ * @returns
+ */
+function evaluatePartsInExpressionChain(
+    parts: string[],
+    operator: ChainingOperator
+): void | ChainedExpression {
+    // Process each part recursively and chain them with ||
+    const firstPart = getExpressionChain(parts[0]);
+    if (firstPart) {
+        let current = firstPart;
+
+        for (let i = 1; i < parts.length; i++) {
+            const nextPart = getExpressionChain(parts[i]);
+            if (nextPart) {
+                // Find the end of the current chain
+                while (current.next) {
+                    current = current.next;
+                }
+                current.next = {
+                    operator,
+                    ...nextPart,
+                };
+            }
+        }
+
+        return firstPart;
+    }
+}
+
 /**
  * Gets the expression chain as a configuration object
  * @param value - The binding string value
  * @returns - A configuration object containing information about the expression
  */
 export function getExpressionChain(value: string): ChainedExpression | void {
-    const split = value.split(" ");
-    let expressionString: string = "";
-    let chainedExpression;
-
-    // split expressions by chaining operators
-    split.forEach((splitItem, index) => {
-        if (splitItem === "&&" || splitItem === "||" || splitItem === "&amp;&amp;") {
-            chainedExpression = {
-                expression: getExpression(expressionString),
-                next: {
-                    operator: splitItem,
-                    ...getExpressionChain(split.slice(index + 1).join(" ")),
-                },
-            };
-        } else {
-            expressionString = `${
-                expressionString ? `${expressionString} ` : ""
-            }${splitItem}`;
+    // Handle operator precedence: || has lower precedence than &&
+    // First, split by || (lowest precedence)
+    const orParts = value.split(" || ");
```

Comment:
Are the spaces around the operators a requirement?

Reply:
Yes, we don't allow `{{foo||bar}}` at this time, the syntax evaluation may get more relaxed in future but right now we're being strict on it.